Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Buttons/ButtonConfigHelper.cs

```diff
@@ -0,0 +1,420 @@
+            }
+
+            if (iconCharLabel == null)
+            {
+                Debug.LogWarning("No icon char label in " + name + " - not setting custom icon char.");
+                return;
+            }
+
+            iconChar = newIconChar;
+            if (iconCharFont != null)
+            {
+                iconCharLabel.font = iconCharFont;
+            }
+
+            uint labelChar = ButtonIconSet.ConvertCharStringToUInt32(iconCharLabel.text);
+            if (labelChar != iconChar || iconCharLabel.font != iconCharFont)
+            {
+                iconCharLabel.text = ButtonIconSet.ConvertUInt32ToUnicodeCharString(newIconChar);
+            }
+
+            SetIconStyle(ButtonIconStyle.Char);
+        }
+
+        /// <summary>
+        /// Sets the sprite for the button. This automatically sets the button icon style to Sprite.
+        /// </summary>
+        public void SetSpriteIcon(Sprite newIconSprite)
+        {
+            if (newIconSprite == null)
+            {
+                return;
+            }
+
+            if (iconSpriteRenderer == null)
+            {
+                Debug.LogWarning("No icon sprite renderer in " + name + " - not setting custom icon sprite.");
+                return;
+            }
+
+            iconSprite = newIconSprite;
+
+            if (iconSpriteRenderer.sprite != iconSprite)
+            {
+                iconSpriteRenderer.sprite = newIconSprite;
+            }
+
+            SetIconStyle(ButtonIconStyle.Sprite);
+        }
+
+        /// <summary>
+        /// Sets the quad texture for the button. This automatically sets the button icon style to Quad.
+        /// </summary>
+        public void SetQuadIcon(Texture newIconTexture)
+        {
+            if (newIconTexture == null)
+            {
+                return;
+            }
+
+            if (iconQuadRenderer == null)
+            {
+                Debug.LogWarning("No icon quad renderer in " + name + " - not setting custom icon texture.");
+                return;
+            }
+
+            iconQuadTexture = newIconTexture;
+
+            if (iconTexturePropertyBlock == null)
+            {
+                iconTexturePropertyBlock = new MaterialPropertyBlock();
+            }
+
+            iconQuadRenderer.GetPropertyBlock(iconTexturePropertyBlock);
+            iconTexturePropertyBlock.SetTexture(iconQuadTextureNameID, newIconTexture);
+            iconQuadRenderer.SetPropertyBlock(iconTexturePropertyBlock);
+
+            SetIconStyle(ButtonIconStyle.Quad);
+        }
+
+        /// <summary>
+        /// Sets the icon style for the button. Relevant components will be turned on / off based on style.
+        /// </summary>
+        private void SetIconStyle(ButtonIconStyle newStyle)
+        {
+            iconStyle = newStyle;
+            switch (iconStyle)
+            {
+                case ButtonIconStyle.Char:
+                    iconCharLabel?.gameObject.SetActive(true);
```

Comment:
Possible incorrect Unity object null checking here. Looks like this label and the renderer references are base `UnityEngine.Object`s, which [don't support](https://docs.unity3d.com/ScriptReference/Object.html) the null conditional operator.

Reply:
These components all inherit from MonoBehaviour (which extends Object) so they should be fine.